### PR TITLE
New Julia Cheat Sheet

### PR DESCRIPTION
**Pull request by akaysh**
Instant Answer Page URL
https://duck.co/ia/view/julia_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@akaysh Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

While the team and the community review your submission, we'd like to make sure you look good! If you'd like a free DuckDuckGo tshirt, please send over the following info to merch@duckduckgo.com, and be sure to put "Dev Merch" in the subject line:

Name:
Street Address:
City:
State / Province:
Country:
Phone (Non-US Orders only): 
T-shirt size:
GitHub username:
Link to this Pull Request:

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/cheat_sheets/json/julia.json [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+    "description": "Julia & IJulia Cheat-sheet (for 18.xxx at MIT)",
+
+    "metadata": {
+        "sourceName": "Math at MIT",
+        "sourceUrl" : "https://www-math.mit.edu/~stevenj/Julia-cheatsheet.pdf"
+    },
+
+    "aliases": [
+        "julia-cheatsheet", "julialang programming"
+    ],
```
Comment (pnpninja) [inline share/goodie/cheat_sheets/json/julia.json]: "julia-cheatsheet" is dubious. remove it

Comment (akaysh) [inline share/goodie/cheat_sheets/json/julia.json]: why am I getting travis-yml fail build?

Comment (akaysh) [inline share/goodie/cheat_sheets/json/julia.json]: It is fine now I think, after the final commit.

Comment (pnpninja) [inline share/goodie/cheat_sheets/json/julia.json]: Congrats on the free shirt :8ball:

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=julia cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/julia_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/julia.json [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+    "aliases": [
+        "julia cheatsheet", "julialang programming"
+    ],
+
+    "template_type": "code",
+
+    "section_order": [
+        "Basics",
+        "Defining\/Changing variables",
```
Comment (mohan08p) [inline share/goodie/cheat_sheets/json/julia.json]: Kindly change the case for variable. Use capital. Variables.. Also at respective section body.

File: share/goodie/cheat_sheets/json/julia.json [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+                "key": "x * 3, x + 3",
+                "val": "Multiply/Add every element of x by 3"
+            },
+            {
+                "key": "x + y",
+                "val": "Element-wise addition of two vectors x and y"
+            },
+            {
+                "key": "A*y, A*B",
+                "val": "product of matrix A and vector y or matrix B (not defined for two vectors)"
```
Comment (mohan08p) [inline share/goodie/cheat_sheets/json/julia.json]: All value fields start with capitals... Product..

File: share/goodie/cheat_sheets/json/julia.json [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+                "key": "x .^ 3",
+                "val": "Every element of x is cubed"
+            },
+            {
+                "key": "cos(x), cos(A)",
+                "val": "Cosine of every element of x or A"
+            },
+            {
+                "key": "exp(A), expm(A)",
+                "val": "exp of each element of A, matrix exp e^A"
```
Comment (mohan08p) [inline share/goodie/cheat_sheets/json/julia.json]: All value fields start with capitals...

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=julia cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/julia_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=julia cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/julia_cheat_sheet_mobile.png)

**Comment by pnpninja** [comment]
Just a note - if its a cheatsheet, shouldn't it display cheatsheet instead of answer?
untitled

**Comment by moollaza** [comment]
@pnpninja fwiw this image is broken too -- how are you uploading them?

File: share/goodie/cheat_sheets/json/julia.json [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+        "Matrices and Vectors",
+        "Arithmetic of Numbers",
+        "Arithmetic of Vectors and Matrices",
+        "Plotting"
+    ],
+
+    "sections": {
+        "Basics": [
+            {
+                "key": "julialang.org",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/julia.json]: Let's ditch the first 3 links -- they don't really fit with the rest of the information here and we don't put that into the other cheat sheets.

I think there may be value in added a `links` property to cheat sheets to let them display some relevant link under the description though. I'll speak to the team.

File: share/goodie/cheat_sheets/json/julia.json [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+
+        ],
+        "Defining\/Changing variables": [
+            {
+                "key": "x=3",
+                "val": "Define variable x to be 3"
+            },
+            {
+                "key": "x=\\[1,2,3\\]",
+                "val": ""
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/julia.json]: Please provide a descriptino

File: share/goodie/cheat_sheets/json/julia.json [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+    "name": "Julia",
+    "description": "Julia & IJulia Cheat-sheet (for 18.xxx at MIT)",
+
+    "metadata": {
+        "sourceName": "Math at MIT",
+        "sourceUrl" : "https://www-math.mit.edu/~stevenj/Julia-cheatsheet.pdf"
+    },
+
+    "aliases": [
+        "julia cheatsheet", "julialang programming"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/julia.json]: The first alias should be removed "julia" is the default "alias" and "cheatsheet" is a default trigger.

Adding this as an alias means "julia cheatsheet cheat sheet" would be the trigger.

What you want is: `"aliases": [ "julialang", "ijulia" ]`

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=julia cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/julia_cheat_sheet_mobile.png)

**Comment by moollaza** [comment]
@akaysh This LGTM :+1:

Thanks again for taking the time to contribute! We really appreciate it :smiley:

**Comment by moollaza** [comment]
@akaysh thanks again for creating the Julia Cheet Sheet! You’ve done a really good job here and we'd like you to help make DuckDuckGo even better for yourself and others interested in the Julia Language.

When you're ready to move on, please: 

- Create a new Instant Answer for searches you do related to Julia Lang

OR

- Test live Julia Instant Answers (or those in development) for bugs and suggest improvements

Detailed instructions explaining how you can help are available at the top of this [page](https://github.com/duckduckgo/duckduckgo/wiki/Creating-or-Improving-Instant-Answers).

Thanks again!

**Pull request merged**